### PR TITLE
chore: update release script

### PR DESCRIPTION
**Pull request by eunjae-lee**
## Summary

This PR updates the release script to align with [the convention](https://developers.google.com/tag-manager/templates/gallery) from Google Tag Manager.

### CODE REVIEW

File: LICENSE [hunk 1]
```diff
@@ -0,0 +1,201 @@
+                                 Apache License
```
Comment (Haroenv) [inline LICENSE]: How come not mit?

Comment (francoischalifour) [reply LICENSE]: That's a [restriction from Google](https://developers.google.com/tag-manager/templates/gallery#prepare_your_project_files).

File: scripts/release.js [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    .trim();
+}
+
+function updateMetadata(commitHash) {
+  const filePath = path.resolve('metadata.yaml');
+  const yaml = YAML.parse(fs.readFileSync(filePath).toString());
+  yaml.versions = [
+    {
+      sha: commitHash,
+      changeNotes: 'Update the template',
```
Comment (francoischalifour) [inline scripts/release.js]: Wdyt about getting the commit title here?

Comment (eunjae-lee) [reply scripts/release.js]: wdyt https://github.com/algolia/search-insights-gtm/pull/1/commits/5844689c5f091936ac8e42f75d6a9c421d04c5ed ?

File: scripts/release.js [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+}
+
+function commitMetadata() {
+  exec(`git add metadata.yaml`);
+  exec(`git commit -m "chore: update metadata.yml"`);
+}
+
+function main() {
+  const { title, hash } = getLatestCommit();
+  updateMetadata(title, hash);
```
Comment (francoischalifour) [inline scripts/release.js]: `updateMedata` expects an object.

```suggestion
  updateMetadata({ title, hash });
```

**Comment by eunjae-lee** [comment]
> Good enough since you're required to push yourself, you can update the release title, right?

Yes, as mentioned in README.md, the flow will be

```
yarn build
git add template.tpl
git commit -m "whatever the release title is"
yarn release # this creates a commit to update metadata.yml
```

**Pull request merged**